### PR TITLE
Tests: Added more specific error messages to the duplicate id asserts

### PR DESCRIPTION
**Pull request by Jarno458**
## What is this fixing or adding?
Added additional comparisations to the duplicate id's check to actually print out the duplicated ids

## How was this tested?
ran them

### CODE REVIEW

**Comment by alwaysintreble** [comment]
Every single case where you added the game name already says the relevant game in the test run.

**Comment by Jarno458** [comment]
i had a hard time finding it

**Comment by alwaysintreble** [comment]
in pycharm
Screenshot_76
on github
Screenshot_77

File: test/general/test_ids.py [hunk 1]
```diff
@@ -24,45 +24,59 @@ def test_range_items(self):
...
                 else:
                     for location_id in world_type.location_id_to_name:
-                        self.assertGreater(location_id, 0)
+                        self.assertGreater(location_id, 0, f"Game: {gamename}")
 
     def test_duplicate_item_ids(self):
         """Test that a game doesn't have item id overlap within its own datapackage"""
         for gamename, world_type in AutoWorldRegister.world_types.items():
             with self.subTest(game=gamename):
-                self.assertEqual(len(world_type.item_id_to_name), len(world_type.item_name_to_id))
```
Comment (alwaysintreble) [inline test/general/test_ids.py:62]:
```suggestion
                self.assertEqual(len(world_type.item_id_to_name), len(world_type.item_name_to_id), set(world_type.item_id.to_name.keys()).difference(set(world_type.item_name_to_id.values()) or set(world_type.item_id.to_name.values()).difference(set(world_type.item_name_to_id.keys()))
```

**Comment by Jarno458** [comment]
seems like my test runner doesnt display those subtests like that

**Comment by alwaysintreble** [comment]
https://pypi.org/project/pytest-subtests/

**Comment by Jarno458** [comment]
Mine doesn't allow me to go to that level, it does still run the test for all games 1 by 1 just fine tho
image

**Comment by Berserker66** [comment]
get a better test runner? No idea which one that even is.

**Comment by Jarno458** [comment]
I dont think adding those game names hurt, might help people who use subpar test runners
but i am required to remove them again i can, let me know if its required to remove them

**Comment by Jarno458** [comment]
Updated to remove the GameName messages

File: test/general/test_ids.py [hunk 2]
```diff
@@ -59,10 +59,24 @@ def test_duplicate_item_ids(self):
...
+                overlap = False
+                if len(world_type.item_id_to_name) != len(world_type.item_name_to_id):
+                    overlap = \
+                        set(world_type.item_id_to_name.values()) \
+                            .symmetric_difference(set(world_type.item_name_to_id.keys())) or \
+                        set(world_type.item_id_to_name.keys()) \
+                            .symmetric_difference(set(world_type.item_name_to_id.values()))
+                self.assertFalse(overlap)
```
Comment (agilbert1412) [inline test/general/test_ids.py:69]: Not a big fan of this style of creating boolean values and asserting them, when the real check is actually the if. What was wrong with the suggested code above?

**Comment by Jarno458** [comment]
Oke lets close this i guess

**Pull request closed without merging**